### ISSUE TITLE
Missing yarn.lock file for vercel deployment

### ISSUE THREAD

**Issue by imjakechapman** [vercel/turborepo#197]
### What version of Turborepo are you using?

1.0.6

### Describe the Bug

I may be missing something here, but followed the getting started on website. I already had a yarn workspaces repo setup with the `/apps/*` and `/packages/*` workspaces. So continued on with setting up the pipelines.

Pipeline:
```
"turbo": {
    "baseBranch": "origin/main",
    "pipeline": {
      "build": {
        "dependsOn": [
          "^build" // prisma package is using dataproxy setup, so I have to make sure the correct bindings are created using the proxy engine and migrate the database.
        ]
      },
      "lint": {},
      "dev": {
        "dependsOn": [
          "^dev" // Since one of my packages is prisma, when I start dev I want to generate latest prisma typings and push the latest db schema.
        ],
        "cache": false
      }
    }
  }
```

In vercel I am using `/apps/platform` for the directory to be deployed and for custom build script I'm using
```
"scripts": {
   "vercel-deploy": "turbo run build",
}
```

Everything locally when I run `yarn dev` from root directory works great, I see cache miss and `packages/prisma` dev command is ran prior to `apps/platform`'s dev command. Got it all committed and pushed up to main to kick of vercel deployment and run into the following. But if I go into `/apps/platform` and run the deploy command I get the following error
```
platform git:(main) ✗ yarn run vercel-deploy
yarn run v1.22.5
$ turbo run build
readfile
 ERROR  yarn.lock: yarn.lock: open yarn.lock: no such file or directory
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Similar error on Vercel deployment
```
[4/4] Building fresh packages...
--
13:36:24.790 | Done in 30.68s.
13:36:24.811 | Detected Next.js version: 12.0.4
13:36:25.003 | Running "yarn run vercel-deploy"
13:36:25.278 | yarn run v1.22.17
13:36:25.305 | $ turbo run build
13:36:25.319 | readfile
13:36:25.319 | ERROR  yarn.lock: yarn.lock: open yarn.lock: no such file or directory
13:36:25.332 | error Command failed with exit code 1.
13:36:25.333 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
13:36:25.344 | Error: Command "yarn run vercel-deploy" exited with 1
```

### Expected Behavior

I'm actually not quite sure. I'm not extremely experienced with monorepos since they've typically already been handled on the job by tooling teams. It appears there's only 1 `yarn.lock` file at the root of monorepo. When I delete that and run yarn install it's recreated. But if I yarn install inside any of the `packages/*` or `apps/*` no yarn.lock is generated. In Vercel I also have the `Include source files outside of the Root Directory in the Build Step.` checkbox checked (not sure if that is causing anything)

### To Reproduce

Simply push main to origin and let Vercel run the custom build script `turbo run build` from within a monorepo directory.

**Comment by jaredpalmer** [comment]
You can follow this guide in the Vercel documentation: https://vercel.com/docs/concepts/git/monorepos

We are actively working on making this experience a bit more seamless.

**Comment by imjakechapman** [comment]
Thanks for the link. Not going to continue this here since I think the next issue is caused via Vercel and not turbo repo but just a heads up currently the follow instructions on that link don't actually work when followed to a T. Hopefully vercel support can take it from here. Thanks again.

You can't set Root Directory to `./apps` and using just `apps` doesn't work. Build takes about 3 seconds and just throws a 404.

<img width="597" alt="Screen Shot 2021-12-11 at 2 43 48 PM" src="https://user-images.githubusercontent.com/1276320/145693859-ed9e243e-c8b8-4148-b0c8-0a0956264629.png">

**Comment by imjakechapman** [comment]
Wanted to leave a quick update since the directions on Vercel's website wasn't working. This also was a little bit of a workaround since I'm using Prisma as dep inside of the `packages/prisma` directory.

1. Follow instructions on vercel's website up until setting Root Directory.
2. For root directory I had to set it like normal `apps/platform`
3. For the build command I had to update it to `cd .. && cd .. && npx turbo run build --scope=<name> --includeDependencies --no-deps`
4. Because I'm using the dataproxy for Prisma the build command for `packages/prisma` wasn't getting access to env variables. To fix this I reverted back to my previous build command  for `apps/platform`
```
"build": "yarn run prisma:deploy && yarn run prisma:generate && next build",
    "prisma:deploy": "DATABASE_URL=\"$DATABASE_MIGRATION_URL\" yarn workspace ${scope} run build:deploy",
    "prisma:generate": "yarn workspace ${scope} run build:generate",
```

At first I thought I could get this working just by using Turborepo's pipeline but now I'm currently not using Turborepo for anything at all since the build command is doing exactly what it was before to read env variables for using the correct production env on Vercel. At first I thought adding `DATABASE_URL=\"$DATABASE_MIGRATION_URL\"` to the `npx turbo run build --scope=<name>` command would do the trick but the `$DATABASE_MIGRATION_URL` config which is set on vercel was not being read.

Again, not sure if this is something Turborepo should be handling or if this is on Vercel. Don't feel like bugging ya if this has nothing to do with Turborepo, may also be something I need to bring up with the Prisma team.
